service/controller/v13/encrypter/tls.go

```diff
@@ -0,0 +1,151 @@
+package encrypter
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/base64"
+
+	"github.com/giantswarm/certs/legacy"
+	"github.com/giantswarm/microerror"
+)
+
+type TLSassets struct {
+	APIServerCA       []byte
+	APIServerKey      []byte
+	APIServerCrt      []byte
+	WorkerCA          []byte
+	WorkerKey         []byte
+	WorkerCrt         []byte
+	ServiceAccountCA  []byte
+	ServiceAccountKey []byte
+	ServiceAccountCrt []byte
+	CalicoClientCA    []byte
+	CalicoClientKey   []byte
+	CalicoClientCrt   []byte
+	EtcdServerCA      []byte
+	EtcdServerKey     []byte
+	EtcdServerCrt     []byte
+}
+
+// PEM encoded TLS assets.
+type RawTLSAssets TLSassets
+
+// Encrypted PEM encoded TLS assets
+type EncryptedTLSAssets TLSassets
+
+func CreateRawTLSAssets(assets legacy.AssetsBundle) *RawTLSAssets {
+	// TODO refactor this with a for loop iterating over components and asset types
```

Comment:
I think this will be different anyway when getting rid of the legacy branch. We just have to switch to master.